code/src/pages/dashboard/dashboard-update-product/DashboardUpdateProductPage.tsx

```diff
@@ -0,0 +1,58 @@
+import { useParams } from "react-router-dom";
+
+import DashboardTabContainer from "@/layouts/dashboard-layout/components/dashboard-tab-container/DashboardTabContainer";
+
+import PageLoadingFallback from "@/containers/page-loading-fallback/PageLoadingFallback";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import useErrorPageRedirect from "@/hooks/use-error-page-redirect/useErrorPageRedirect";
+import { updateProductPageNotFoundErrorConfig } from "@/pages/dashboard/dashboard-update-product/DashboardUpdateProductPage.constants";
+import { useGetManagerProductQuery } from "@/store/api/productsApi";
+import isErrorWithStatus from "@/utils/is-error-with-status/isErrorWithStatus";
+import isUuidValid from "@/utils/is-uuid-valid/isUUIDValid";
+
+import "@/pages/dashboard/dashboard-update-product/DashboardUpdateProductPage.scss";
+
+const DashboardUpdateProductPage = () => {
+  const { productId } = useParams();
+
+  const { renderRedirectComponent } = useErrorPageRedirect();
+
+  if (!productId || !isUuidValid(productId)) {
+    return renderRedirectComponent(updateProductPageNotFoundErrorConfig);
+  }
+
+  const { data, isLoading, error } = useGetManagerProductQuery({ productId });
```

Comment:
Did you test this component? I am not sure but I think we can get error react rendered more hook camparing to previous render because we try to use hook after return

Reply:
I believe currently it woudn't throw error because it depends on id in route and if we change id through browser search bar, entire page will be reloadeded. In other words, there are no possibility that the same component will be firstly rendered with that hook and after without it. But in future, there may be problems if we will add some additional routing logic to this page. I changed it, now there are two components